File: api/src/auth/auth.controller.ts
import { Controller, Post, Get, Query, Body } from '@nestjs/common';
import { AuthService } from './auth.service';
import { CreateUserDto } from '../user/dto/create-user.dto';
import { CheckNicknameDto } from './dto/check-nickname.dto';

@Controller('auth')
export class AuthController {
  constructor(private readonly authService: AuthService) {}

  @Post('register')
  async register(@Body() createUserDto: CreateUserDto) {
    return this.authService.register(createUserDto);
  }

  @Post('login')
  async login(@Body() body: { email: string; password: string }) {
    return this.authService.login(body.email, body.password);
  }

  @Get('check-nickname')
  async checkNickname(@Query() dto: CheckNicknameDto) {
    return this.authService.checkNicknameAvailable(dto.nickname);
  }
}

File: api/src/main/java/com/chatops/domain/auth/controller/AuthController.java
package com.chatops.domain.auth.controller;

import com.chatops.domain.auth.dto.CreateUserRequest;
import com.chatops.domain.auth.dto.LoginRequest;
import com.chatops.domain.auth.dto.LoginResponse;
import com.chatops.domain.auth.dto.UserResponse;
import com.chatops.domain.auth.service.AuthService;
import jakarta.validation.Valid;
import jakarta.validation.constraints.Size;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/auth")
@RequiredArgsConstructor
public class AuthController {

    private final AuthService authService;

    @PostMapping("/register")
    public UserResponse register(@Valid @RequestBody CreateUserRequest request) {
        return authService.register(request);
    }

    @PostMapping("/login")
    public LoginResponse login(@Valid @RequestBody LoginRequest request) {
        return authService.login(request.getEmail(), request.getPassword());
    }

    @GetMapping("/check-nickname")
    public Map<String, Boolean> checkNickname(@RequestParam @Size(min = 2, max = 20) String nickname) {
        return Map.of("available", authService.checkNicknameAvailable(nickname));
    }
}
